packages/widgetbook/test/src/addons/custom_theme_addon_test.dart

```diff
@@ -0,0 +1,210 @@
+                'This is the home page',
+              ),
+            ],
+          ),
+        ),
+      ),
+    );
+  });
+
+  group('$Renderer', () {
+    testWidgets(
+      'renders correctly',
+      (WidgetTester tester) async {
+        await tester.pumpWidget(
+          addOnProviderWrapper(
+            child: sut,
+          ),
+        );
+
+        expect(find.byKey(rendererKey), findsOneWidget);
+      },
+    );
+    testWidgets(
+      'renders one Text Widget',
+      (WidgetTester tester) async {
+        await tester.pumpWidget(
+          addOnProviderWrapper(
+            child: sut,
+          ),
+        );
+
+        expect(find.byType(Text), findsOneWidget);
+      },
+    );
+
+    testWidgets(
+      'renders the correct text',
+      (WidgetTester tester) async {
+        await tester.pumpWidget(
+          addOnProviderWrapper(
+            child: sut,
+          ),
+        );
+
+        expect(find.text('This is the home page'), findsOneWidget);
+      },
+    );
+    testWidgets(
+      'renders the correct theme',
+      (WidgetTester tester) async {
+        await tester.pumpWidget(
+          addOnProviderWrapper(
+            child: sut,
+          ),
+        );
+        await tester.pumpAndSettle();
+
+        final scaffoldFinder = find.byType(Scaffold);
+        expect(scaffoldFinder, findsOneWidget);
+        final scaffold = scaffoldFinder.evaluate().single.widget as Scaffold;
+
+        expect(scaffold.backgroundColor, equals(customColor));
+      },
+    );
+
+    testWidgets(
+      'renders theme side by side when 2 active themes are activated',
+      (WidgetTester tester) async {
+        final provider = ThemeSettingProvider(
+          CustomThemeSetting<AppThemeData>(
+            activeThemes: {customTheme},
+            themes: [customTheme, customTheme2],
+          ),
+        );
+
+        await tester
+            .pumpWidget(addOnProviderWrapper(child: sut, provider: provider));
+        provider.tapped(customTheme2);
+
+        await tester.pumpAndSettle();
+
+        final scaffoldFinderUpdated = find.byType(Scaffold);
+        expect(scaffoldFinderUpdated, findsWidgets);
+        final updatedScaffoldColors = scaffoldFinderUpdated.evaluate().map((e) {
+          final scaffold = e.widget as Scaffold;
+          return scaffold.backgroundColor;
+        }).toList();
+
+        expect(
+          updatedScaffoldColors.length,
+          equals(
+            2,
+          ),
+        );
+
+        expect(
+          updatedScaffoldColors,
+          equals(
+            [
+              customColor,
+              customColor2,
+            ],
+          ),
+        );
+      },
+    );
+
+    testWidgets(
+      'can de-activate $customTheme2 when $ThemeSettingProvider..tapped(customTheme2) is called',
+      (WidgetTester tester) async {
+        final provider = ThemeSettingProvider(
+          CustomThemeSetting<AppThemeData>(
+            activeThemes: {customTheme},
```

Comment:
you should have customTheme2 as an active theme here because otherwise you'll also test activating. However, this test is called "de-activate". In addition, you'd also cover the test behavior of the "activate" test with this one. Why do we need the other test at all if this covers the functionality as well? The reason is that we want to test activate and deactivate separately.